### PR TITLE
Remove jargon from and simplify README

### PR DESCRIPTION
**Pull request by benhoyt**
**[View rendered version updated README.md here.](https://github.com/benhoyt/operator/blob/readme-update/README.md)**

Remove a lot of the "operator" and K8s jargon from the top of our README. I feel fairly strongly this isn't the place to sell how great Juju, K8s operators, and model-driven development are -- we should do that on our landing page and/or docs, not in the ops README. Charmers shouldn't normally be landing here, and if they do, I think we should give them succinct, developer-oriented documentation.

Also simplify and update some of the terminology to be consistent with our updated Juju Terminology doc.

The first part of the README is copied to `ops/__init__.py` for consistency. That will appear at the top of our API reference at https://ops.readthedocs.io/

Move the note about dependencies to HACKING.md where I think it belongs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,103 +1,55 @@
 # The Charmed Operator Framework
 
-This Charmed Operator Framework simplifies [operator](https://charmhub.io/about) development
-for [model-driven application management](https://juju.is/model-driven-operations).
+The Charmed Operator Framework is a Python library (available as [ops](https://pypi.org/project/ops/))
```
Comment (jnsgruk) [inline README.md:3]:
```suggestion
The Operator Framework is a Python library (available as [ops](https://pypi.org/project/ops/))
```

Based upon recent changes in the Juju Terminology doc and recent conversations :) applies throughout

File: README.md [hunk 2]
```diff
@@ -1,103 +1,55 @@
...
+for developing and testing [Juju](https://juju.is/) charms in a consistent way, using standard Python constructs
+to allow for clean, maintainable, and reusable code.
 
-Operators emerged from the Kubernetes community; an operator is software that drives lifecycle
-management, configuration, integration and daily actions for an application. Operators simplify
-software management and operations. They capture reusable app domain knowledge from experts in a
-software component that can be shared.
+Charms written with the Charmed Operator Framework support Kubernetes using
```
Comment (jnsgruk) [inline README.md:7]: I think majoring on "charms" is the right way to go, but I'd include this (short) definition/clarification in the README too:

```
A charm is an operator - business logic encapsulated in reusable software packages that automate every aspect of an application's life. 
```

File: README.md [hunk 3]
```diff
@@ -1,103 +1,55 @@
...
-Operators use a [Charmed Operator Lifecycle Manager
-(Charmed OLM)](https://juju.is/operator-lifecycle-manager) to coordinate their work in a cluster.
-The system uses Golang for concurrent event processing under the hood, but enables the operators to
-be written in Python.
+The Charmed Operator Framework is part of the Charm SDK (the other part being
+Charmcraft). Full developer documentation for the Charm SDK is available at
+https://juju.is/docs/sdk.
 
-## Simple, composable operators
+To learn more about the Juju engine, visit https://juju.is/docs/olm.
```
Comment (jnsgruk) [inline README.md:21]:
```suggestion
To learn more about Juju, visit https://juju.is/docs/olm.
```

File: README.md [hunk 4]
```diff
@@ -1,103 +1,55 @@
...
+for developing and testing [Juju](https://juju.is/) charms in a consistent way, using standard Python constructs
+to allow for clean, maintainable, and reusable code.
 
-Operators emerged from the Kubernetes community; an operator is software that drives lifecycle
-management, configuration, integration and daily actions for an application. Operators simplify
-software management and operations. They capture reusable app domain knowledge from experts in a
-software component that can be shared.
+Charms written with the Charmed Operator Framework support Kubernetes using
+Juju's "sidecar charm" pattern, as well as charms that deploy to Linux-based
+virtual machines.
```
Comment (jnsgruk) [inline README.md:9]:
```suggestion
machines and containers.
```

File: README.md [hunk 5]
```diff
@@ -1,103 +1,55 @@
...
 
-This project extends the operator pattern to enable
-[charmed operators](https://juju.is/universal-operators), not just for Kubernetes but also
-operators for traditional Linux application management.
+Charms should do one thing and do it well. Each charm drives a single Juju
```
Comment (jnsgruk) [inline README.md:11]:
```suggestion
Charms should do one thing and do it well. Each charm drives a single
```

**Comment by benhoyt** [comment]
@jnsgruk Thanks for the review -- updates made, ready for re-review.

**Comment by benhoyt** [comment]
Merged. Note posted on Charmhub Mattermost > Charm Development, and also at https://discourse.charmhub.io/t/terminology-tweak-operator-framework-ops/10908

**Pull request merged**